\documentclass[oneside, a4paper]{memoir}
\renewcommand{\rmdefault}{ptm}
\title{Assessing the problem of doing proteomics with \mbox{unsequenced organisms}\\
\textsc{\small Lab Rotation project in Bioinformatics}
}
\author{
	        R\'oger Berm\'udez-Chac\'on\thanks{D-INFK - Computational Biology and Bioinformatics Master program}\\
			 Supervisor: Katja B\"arenfaller\thanks{D-BIOL - Plant Biotechnology Group}
}
\date{June 10, 2013}

\setlrmarginsandblock{3cm}{2.5cm}{*}
\setulmarginsandblock{2.5cm}{2.5cm}{*}
\checkandfixthelayout

\begin{document}
\maketitle
\setcounter{secnumdepth}{0}

\section{Project description}
When dealing with unannotated or partially annotated proteomes, protein identif{}ication is often carried out by
%homology comparison against a reference species (or a set thereof).
searching a database containing protein sequences from other species.
%Homology inference by correspondence or similarity between ion spectra is thus
Identif{}ication of a protein from a different species is
often assumed as enough evidence to 
%label 
claim that this protein was identif{}ied in
a protein sample.
This work attempts to systematically evaluate at what extent this holds true, by comparing search results of unannotated
proteins extracted from the Cassava \mbox{(\emph{Manihot~esculenta})} root against dif{}ferent annotated databases, including
the reference species \mbox{\emph{Arabidopsis~thaliana}}, the Viridiplantae~database (containing a large number of green plants), and the~annotated~proteome of Cassava itself.
\end{document}
